src/Spago/Command/Fetch.purs

```diff
@@ -404,42 +404,41 @@ getTransitiveDepsFromPackageSet packageSet deps = do
     init :: TransitiveDepsResult
     init = { packages: (Map.empty :: Map PackageName Package), errors: mempty }
 
-    mergeResults :: TransitiveDepsResult -> TransitiveDepsResult -> TransitiveDepsResult
-    mergeResults r1 r2 =
-      { packages: Map.union r1.packages r2.packages
-      , errors: r1.errors <> r2.errors
-      }
-
-    go :: Set PackageName -> PackageName -> StateT (Map PackageName (Map PackageName Package)) (Spago (FetchEnv a)) TransitiveDepsResult
-    go seen dep =
-      if (Set.member dep seen) then do
-        pure (init { errors { cycle = Set.singleton dep } })
+    -- | Cheap size check to be used in place of Set equality
+    errorSize :: TransitiveDepsResult -> Int
+    errorSize { errors: { notInPackageSet: p, notInIndex: i, cycle: c } } =
+      Set.size p + Set.size i + Set.size c
```

Comment:
Could you expand (in the docstring) on the reasoning for why this exists?

Reply:
On second thought this is almost definitely preemptive optimisation. Probably actually makes sense to just check equality of the whole errors record. 

That way if new errors are added in the future there won't be a worry about having to update this. That or use heterogeneous to get the size of all error `Set`s in the record (I just didn't want to add an extra dep to spago).

Reply:
I'll add a comment or replace with equality this evening :+1:

Reply:
Let's try with Set equality and see how slow it gets.

If that doesn't work then my next favourite option would be to replace the record that sorts errors in different buckets with just a List/Array/Set of `TransitiveDepsError` (which would be a sum type of all the errors we could get). Then we could check for the size of that.